client/src/modules/department/department.ctrl.js

```diff
@@ -0,0 +1,102 @@
+
+angular.module('bhima.controllers')
+  .controller('DepartementController', DepartementController);
+
+DepartementController.$inject = [
+  'DepartmentService', '$uibModal', 'ModalService',
+  'NotifyService', 'uiGridConstants', 'SessionService',
+];
+
+function DepartementController(Department, Modal, ModalService, Notify, uiGridConstants, Session) {
+  const vm = this;
+  const { enterprise } = Session;
+  // global variables
+  vm.gridApi = {};
+  vm.filterEnabled = false;
+  vm.toggleFilter = toggleFilter;
+  vm.createDepartement = createDepartement;
+  vm.deleteDepartment = deleteDepartment;
+
+  // options for the UI grid
+  vm.gridOptions = {
+    appScopeProvider  : vm,
+    enableColumnMenus : false,
+    fastWatch         : true,
+    flatEntityAccess  : true,
+    enableSorting     : true,
+    onRegisterApi     : onRegisterApiFn,
+    columnDefs : [
+      {
+        field : 'name',
+        displayName : 'FORM.LABELS.NAME',
+        headerCellFilter : 'translate',
+      },
+      {
+        field : 'action',
+        width : 80,
+        displayName : '',
+        cellTemplate : '/modules/department/templates/action.tmpl.html',
+        enableSorting : false,
+        enableFiltering : false,
+      },
+    ],
+  };
+
+  function onRegisterApiFn(gridApi) {
+    vm.gridApi = gridApi;
+  }
+
+  function toggleFilter() {
+    vm.filterEnabled = !vm.filterEnabled;
+    vm.gridOptions.enableFiltering = vm.filterEnabled;
+    vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.COLUMN);
+  }
+
+  // get all enterprise's depatments
+  function loadDepartments() {
+    Department.read(null, { enterprise_id : enterprise.id })
```

Comment:
Hmm.. do you think we will ever need to list all departments, regardless of enterprise?  Most other APIs that use enterprises just attach the `enterprise_id` on the server.